### PR TITLE
🎉 Source Zuora: Add `OAuth` support for Airbyte Cloud, Migrated to new CI Sandbox

### LINKED ISSUE #6035: Use new Zuora Sandbox in CI
## Tell us about the problem you're trying to solve
We have a new zuora sandbox account that we should use in CI. This one is a permanent partner account which will not expire like the last one. 

## Describe the solution you’d like
1. Populate the new account with data (`integration-test+sandbox@airbyte.io`)
2. use it in CI
3. delete the old account from lastpass

### LINKED ISSUE #6298: Source Zuora: support oauth
## Tell us about the problem you're trying to solve
With the release of Airbyte Cloud, we need to start supporting Oauth for this connector, since it's the recommended way of authenticating users into a SaaS application. 

If this connector doesn't support oauth already (i.e: doesn't accept a client_id and client_secret) then we need to update its spec to accept those parameters. There are two ways to do this: 

**If the connector already supports some auth mechanism like api_key**, I suggest that this be a oneof nested inside a top-level field called "authentication": 

`
{
  authentication: {
     type: object
     oneOf: [ 
       // api key,
       // oauth
     ] 
  }
}
`

**If the connector only supports webflow oauth**, then no changes are needed to the properties format and we will only need to add annotations. 

See the connector spec reference in the docs for reference on how a oneof can be implemented.

This should be done in a backwards compatible manner i.e: users currently supplying authentication info in the config's top-level should not be impacted by this change. 

## Acceptance Criteria

1. The connector supports oauth webflow authentication with client_id/client_secret in a backwards compatible manner
2. Oauth properties are annotated properly. See [this PR](https://github.com/airbytehq/airbyte/pull/6022) for an example

### PR DESCRIPTION
**Pull request by bazarnov**
## What
#6298 - Source Zuora: support oauth

## How
- edited `spec.json`
- added more endpoints to connect to (as options in UI)
- migrated to new CI Sandbox (airbyte-integration)
- added support of streams without any cursor (updatedDate, createdDate) - for such cases, we fetch the whole object using Full-Refresh mode.
- edited `integration-test.py`

## Pre-merge Checklist
<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [X] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [X] Documentation updated 
    - [X] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [X] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [X] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-zuora/source_zuora/zuora_endpoint.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "EU Production": "https://rest.eu.zuora.com",
+    "EU API Sandbox": "https://rest.sandbox.eu.zuora.com",
+    "EU Central Sandbox": "https://rest.test.eu.zuora.com",
+}
+
+def get_url_base(tenant_endpoint: str) -> str:
+    """ Define the URL Base from user's input with respect to the ZUORA_TENANT_ENDPOINT_MAP """
+
+    # map the tenant_type & endpoint from user's input
+    tenant_type, endpoint = list(tenant_endpoint.items())[0]
```
Comment (antixar) [inline airbyte-integrations/connectors/source-zuora/source_zuora/zuora_endpoint.py:24]: `tenant_endpoint` is marked as string but this function uses it as dict. `tenant_endpoint.items()`. And this part `list(tenant_endpoint.items())[0]` is not incomprehensible. Could add an example of the tenant_endpoint value as code comment?

**Review by antixar — CHANGES_REQUESTED** [review]
are you sure that you added oauth properties to the spec file? I see changes for sandbox options only.

File: airbyte-integrations/connectors/source-zuora/source_zuora/zuora_endpoint.py (hunk 1 above)
Comment (bazarnov) [reply airbyte-integrations/connectors/source-zuora/source_zuora/zuora_endpoint.py:24]: This option is not valid anymore. It's changed.

File: airbyte-integrations/connectors/source-zuora/source_zuora/spec.json [hunk 2]
```diff
@@ -28,10 +28,49 @@
...
+          {
+            "title": "Sandbox",
+            "description": "Choose between sandbox endpoints",
+            "required": ["sandbox"],
+            "properties": {
+              "sandbox": {
+                "title": "Choose between sandbox endpoints",
+                "type": "string",
+                "enum": ["US API Sandbox","US Cloud API Sandbox","US Central Sandbox","EU API Sandbox","EU Central Sandbox"]
+              }
```
[the remark this answers is not in the document]
Comment (bazarnov) [reply airbyte-integrations/connectors/source-zuora/source_zuora/spec.json:57]: This option is not valid anymore. It's changed.

**Comment by bazarnov** [comment]
> are you sure that you added oauth properties to the spec file? I see changes for sandbox options only.

Haven't manage to submit the latest changes yet) The tests on new CI Sandbox require to update some parts  of the connector.

File: airbyte-integrations/connectors/source-zuora/source_zuora/spec.json [hunk 3]
```diff
@@ -15,7 +19,14 @@
       "window_in_days": {
         "type": "integer",
         "description": "The amount of days for each data-chunk begining from start_date. Bigger the value - faster the fetch. (Min=1, as for a Day; Max=364, as for a Year).",
-        "examples": [30, 60, 90, 120, 200, 364],
+        "examples": [
```
Comment (antixar) [inline airbyte-integrations/connectors/source-zuora/source_zuora/spec.json:22]: add minimum and maximum values for this option

**Review by antixar — APPROVED** [review]
looks good!

File: airbyte-integrations/connectors/source-zuora/source_zuora/spec.json (hunk 3 above)
Comment (bazarnov) [reply airbyte-integrations/connectors/source-zuora/source_zuora/spec.json:22]: Sure, done.

**Pull request merged**